src/storage/src/client/controller.rs

```diff
@@ -484,9 +484,12 @@ where
                 .await
                 .expect("invalid persist usage");
 
-            let collection_state =
-                CollectionState::new(description.clone(), read.since().clone(), metadata);
-
+            let collection_state = CollectionState::new(
```

Comment:
This is not correct and would also need to make sure that the `storaged`s also retract the correct previous upper in the beginning (instead of 0).